x/staking/simulation/operations.go

```diff
@@ -376,60 +376,60 @@ func SimulateMsgUndelegate(ak types.AccountKeeper, bk types.BankKeeper, k keeper
 func SimulateMsgBeginRedelegate(ak types.AccountKeeper, bk types.BankKeeper, k keeper.Keeper) simtypes.Operation {
 	return func(
 		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
-	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, sdk.Result, error) {
 		// get random source validator
 		srcVal, ok := keeper.RandomValidator(r, k, ctx)
 		if !ok {
-			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgBeginRedelegate, "unable to pick validator"), nil, nil
+			return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgBeginRedelegate, "unable to pick validator"), nil, sdk.Result{}, nil
```

Comment:
Asking to learn: 
- where exactly are we adding the result to the data hash?
- Why are most of the `sdk.Result{}` returns empty?

Reply:
> where exactly are we adding the result to the data hash?

I saw some examples in slashing, that makes sense.

> Why are most of the sdk.Result{} returns empty?

It seems to be that many of the returns correspond to errors.

Not necessarily in this PR but I'm wondering if it makes sense to extract a helper for error returns? For example,
```
return simtypes.NoOpMsg(txCtx.ModuleName, txCtx.MsgType, "message doesn't leave room for fees"), nil, sdk.Result{}, err
```

Seems to be repeated many times with a different error message.

Reply:
The data hash result is basically returned in any simulation weightedOperation call. This change was made locally in the osmosis repo, however any legacy messages (i.e. messages that come directly from our sdk fork) needed this change so they could also return the result.

Yeah, any time the result is returned as blank its just an error line, 100% agree this could/should be abstracted to reduce noise